api/src/main/java/org/openmrs/module/eptsreports/reporting/library/datasets/ListOfPatientsCurrentlyOnArtWithoutTbScreeningDataset.java

```diff
@@ -0,0 +1,237 @@
+package org.openmrs.module.eptsreports.reporting.library.datasets;
+
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import org.openmrs.Location;
+import org.openmrs.PatientIdentifierType;
+import org.openmrs.PersonAttributeType;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.eptsreports.metadata.HivMetadata;
+import org.openmrs.module.eptsreports.reporting.data.converter.DispensationTypeConverter;
+import org.openmrs.module.eptsreports.reporting.data.converter.GenderConverter;
+import org.openmrs.module.eptsreports.reporting.data.converter.NotApplicableIfNullConverter;
+import org.openmrs.module.eptsreports.reporting.library.cohorts.*;
+import org.openmrs.module.eptsreports.reporting.library.queries.TxtbDenominatorQueries;
+import org.openmrs.module.reporting.data.DataDefinition;
+import org.openmrs.module.reporting.data.converter.DataConverter;
+import org.openmrs.module.reporting.data.converter.ObjectFormatter;
+import org.openmrs.module.reporting.data.patient.definition.ConvertedPatientDataDefinition;
+import org.openmrs.module.reporting.data.patient.definition.PatientIdentifierDataDefinition;
+import org.openmrs.module.reporting.data.person.definition.ConvertedPersonDataDefinition;
+import org.openmrs.module.reporting.data.person.definition.GenderDataDefinition;
+import org.openmrs.module.reporting.data.person.definition.PersonAttributeDataDefinition;
+import org.openmrs.module.reporting.data.person.definition.PreferredNameDataDefinition;
+import org.openmrs.module.reporting.dataset.definition.DataSetDefinition;
+import org.openmrs.module.reporting.dataset.definition.PatientDataSetDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class ListOfPatientsCurrentlyOnArtWithoutTbScreeningDataset extends BaseDataSet {
+
+  @Autowired
+  private ListOfPatientsCurrentlyOnArtWithoutTbScreeningCohortQueries
+      listOfPatientsCurrentlyOnArtWithoutTbScreeningCohortQueries;
+
+  @Autowired private ListOfPatientsArtCohortCohortQueries listOfPatientsArtCohortCohortQueries;
+
+  @Autowired private TPTListOfPatientsEligibleDataSet tptListOfPatientsEligibleDataSet;
+
+  @Autowired
+  private ListOfPatientsDefaultersOrIITCohortQueries listOfPatientsDefaultersOrIITCohortQueries;
+
+  @Autowired private TPTInitiationDataDefinitionQueries tptInitiationDataDefinitionQueries;
+
+  @Autowired
+  private ListOfPatientsEligibleForVLDataDefinitionQueries
+      listOfPatientsEligibleForVLDataDefinitionQueries;
+
+  @Autowired private TxtbDenominatorQueries txtbDenominatorQueries;
+
+  @Autowired private HivMetadata hivMetadata;
```

Comment:
Collapse all those Autowired calls into one encapsulated in a constructor

Reply:
done